tests/test_udfs.py

```diff
@@ -58,6 +58,27 @@
       ' but got ',
       TO_JSON_STRING(actual))),
     TRUE));
+
+CREATE TEMP FUNCTION
+  assert_sets_equals(expected ANY TYPE, actual ANY TYPE) AS (
```

Comment:
Perhaps this should be `assert_array_equals_any_order` since I think this can support multiple identical entries. I don't think this fully handles the case, though, where multiple different entries appear multiple times, since it only checks the overall array lengths match.

Reply:
Agreed, it's not perfect. We have three options:
1. Ignore duplicate values (essentially, turn the arrays to sets) 
2. Ensure arrays in any order, with dupes, match
3. Fail if any array has dupes (contract that they are sets)

I'd be good with (2.) or (3.). We should avoid (1.) for obvious reasons.

Reply:
I'd recommend that we sort both arrays and then compare that all elements are equal by offset.

Something like:

```
WITH a AS (
  SELECT x FROM UNNEST(arr1) AS x OFFSET AS offset ORDER BY x)
), b AS (
  SELECT x FROM UNNEST(arr2) AS x OFFSET AS offset ORDER BY x)
)
select a join b using (offset)
```

Reply:
I'm good with (2.). I'll update the test.